### ISSUE TITLE
Build fails for projects scaffolded in the past

### ISSUE THREAD

**Issue by agiantwhale** [yeoman/generator-webapp#564]
I have a project that I scaffolded in the past, and at the time building it worked perfectly. I'm trying to compile the project recently (after cloning & npm/bower installing on a new machine), but it's currently failing.

Why is this the case? Could someone look at [my current project](https://github.com/agiantwhale/mfat) and see what is wrong?

```
[08:09:03] Failed to load external module babel-register
[08:09:03] Requiring external module babel-core/register
[08:09:03] Using gulpfile ~/Development/mfat/gulpfile.babel.js
[08:09:03] Starting 'lint'...
[08:09:03] Starting 'styles'...
[08:09:04] Starting 'images'...
[08:09:04] Starting 'fonts'...
[08:09:04] Starting 'extras'...
[08:09:05] Finished 'images' after 891 ms
[08:09:05] Finished 'lint' after 1.29 s
[08:09:05] Finished 'styles' after 1.12 s
[08:09:05] Starting 'html'...

events.js:160
      throw er; // Unhandled 'error' event
      ^
Error
    at new JS_Parse_Error (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:1534:18)
    at js_error (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:1542:11)
    at croak (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2089:9)
    at token_error (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2097:9)
    at unexpected (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2103:9)
    at semicolon (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2123:56)
    at simple_statement (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2314:73)
    at eval (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2183:19)
    at eval (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2136:24)
    at eval (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2904:23)
    at Object.parse (eval at <anonymous> (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:28:1), <anonymous>:2915:7)
    at /home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:71:33
    at Array.forEach (native)
    at Object.exports.minify (/home/iljae/Development/mfat/node_modules/uglify-js/tools/node.js:66:15)
    at /home/iljae/Development/mfat/node_modules/gulp-uglify/minifier.js:65:22
    at trycatch (/home/iljae/Development/mfat/node_modules/gulp-uglify/minifier.js:15:12)
```

I'm creating an issue as I've noticed that this is a recurring for the generator - I've had to scaffold a new web app and copy my app folder to the new scaffolded project multiple times, for several of my projects.

**Comment by neilhem** [comment]
Hi, have you tried to reinstall dependencies?
**P.S.** Looks like you have js syntax error, so uglify task throws error

**Comment by agiantwhale** [comment]
Yes, I am doing this after cloning the repositories on to a new drive.  

Il Jae Lee![](https://link.nylas.com/open/d4c596q61b29wh2t5efvjyecb/local-9f2a
9b48-a059?r=cmVwbHkrMDAyZmRhY2YxZGI5M2RhNTgzOWUxYWQ5NWFlNDA5YjRiZjcwYTdhNzVlOD
k4NjZhOTJjZjAwMDAwMDAxMTNkYzBkYjk5MmExNjljZTBhNWEwYzE1QHJlcGx5LmdpdGh1Yi5jb20=)

**Comment by silvenon** [comment]
Have you tried fixing the syntax error like @neilhem suggested?

**Comment by martisj** [comment]
Which version of node are you running this on?

If it's node v6 try updating `gulp-if`. I had pretty much the same issue. Turns out the gulp-if is not filtering on the

``` javascript
.pipe($.if('*.js', $.uglify()))
```

call so it's passing all files to uglify. Even though uglify is a beast it doesn't parse html.... yet..

Try adding

``` javascript
.pipe($.plumber())
```

on the `html` gulp build task. Right before this line https://github.com/yeoman/generator-webapp/blob/master/app/templates/gulpfile.js#L75

That will catch the error and show you something a little more interesting. I found the solution over here #513 

@silvenon I think that this might be related to this one as well #572 

Hope this helps mr/mrs @agiantwhale
